test/integration/TestHeap.t.sol

```diff
@@ -5,314 +5,319 @@ import {Test} from "forge-std/Test.sol";
 
 import {BasicHeap} from "src/Heap.sol";
 
+import {HeapMock} from "../mocks/HeapMock.sol";
+
 contract TestHeap is Test {
-    using BasicHeap for BasicHeap.Heap;
+    /* STORAGE */
 
-    uint256 public TESTED_SIZE = 50;
-    address[] public accounts;
-    address public ADDR_ZERO = address(0);
+    uint256 public constant TESTED_SIZE = 50;
 
-    BasicHeap.Heap internal heap;
+    HeapMock internal heap;
 
-    function setUp() public {
-        accounts = new address[](TESTED_SIZE);
-        accounts[0] = address(bytes20(keccak256("TestHeap.accounts")));
-        for (uint256 i = 1; i < TESTED_SIZE; i++) {
-            accounts[i] = address(uint160(accounts[i - 1]) + 1);
+    function _createAccounts(bytes32 seed, uint256 n) internal pure returns (address[] memory accounts) {
+        accounts = new address[](n);
+        for (uint256 i; i < n; ++i) {
+            seed = keccak256(abi.encode(seed));
+            accounts[i] = address(bytes20(seed));
         }
     }
 
-    function testInsertOneSingleAccount() public {
-        heap.insert(accounts[0], 1);
-
-        assertEq(heap.length(), 1);
-        assertEq(heap.getValueOf(accounts[0]), 1);
-        assertEq(heap.getRoot(), accounts[0]);
-        assertEq(heap.getLeftChild(accounts[0]), ADDR_ZERO);
-        assertEq(heap.getRightChild(accounts[0]), ADDR_ZERO);
+    function _createIncreasingValues(bytes32 seed, uint256 min, uint256 max, uint256 n)
+        internal
+        view
+        returns (uint256[] memory values)
+    {
+        values = new uint256[](n);
+        uint256 minValue = min;
+        for (uint256 i; i < n; ++i) {
+            seed = keccak256(abi.encode(seed));
+            values[i] = bound(uint256(seed), minValue, max - (n - i));
```

Comment:
This method of generating random increasing values skews heavily towards higher values as n increases. Why not just generate n random numbers and sort them? The reason this should be okay is because:

* n never seems to be a very high value. So even a naive insertion sort would be just fine.
* the possibility of a keccak256 colliding with different inputs is negligible.

Reply:
yes, but it has the benefit of having linear complexity. Sorting the values would take n log(n) time